test/utils/formatting/dataItem.js

```diff
@@ -0,0 +1,255 @@
+import expect from 'expect.js';
+
+import { formatValueFromStateAndProps } from '../../../src/utils/formatting/dataItem';
+
+const createState = ({
+  symbol = 'MSFT',
+  value,
+  optionalValue,
+  valueSuffix,
+  optionalValueSuffix
+}) => ({
+  selectedStock: symbol,
+  stocks: {
+    MSFT: {
+      stockData: {
+        Div: {
+          value,
+          optionalValue,
+          valueSuffix,
+          optionalValueSuffix
+        }
+      }
+    }
+  }
+});
+
+const createProps = ({
+  label,
+  valuePrecision,
+  optionalValuePrecision
+}) => ({
+  label,
+  valuePrecision,
+  optionalValuePrecision
+});
+
+export const formatValueFromStateAndPropsTest = () => {
+  const VALUE = 3.54;
+  const OPTIONAL_VALUE = 0.45;
+  const VALUE_SUFFIX = '%';
+  const OPTIONAL_VALUE_SUFFIX = '%';
+
+  const VALUE_PRECISION = 2;
+  const OPTIONAL_VALUE_PRECISION = 2;
+
+  describe('formatValueFromStateAndProps', () => {
+    it('should properly format when ALL fields are defined', () => {
+      const state = createState({
+        value: VALUE,
+        optionalValue: OPTIONAL_VALUE,
+        valueSuffix: VALUE_SUFFIX,
+        optionalValueSuffix: OPTIONAL_VALUE_SUFFIX
+      });
+      const ownProps = createProps({
+        label: 'Div',
+        valuePrecision: VALUE_PRECISION,
+        optionalValuePrecision: OPTIONAL_VALUE_PRECISION
+      });
+      expect(formatValueFromStateAndProps(state, ownProps)).to
```

Comment:
If it was me, I think I'd put `.to` on its own line as well (same with all cases below).
```javascript
expect(formatValueFromStateAndProps(state, ownProps))
  .to
  .equal(VALUE + VALUE_SUFFIX + ' (' + OPTIONAL_VALUE + OPTIONAL_VALUE_SUFFIX + ')');
```
is just so easy to read (which is the whole point of `.to`'s existence)